aten/src/ATen/native/native_functions.yaml

```diff
@@ -37,19 +37,27 @@
   use_c10_dispatcher: full
   variants: function
 
+# Computes the gradient of current tensor w.r.t. graph leaves.
 - func: backward(Tensor self, Tensor? gradient=None, bool keep_graph=False, bool create_graph=False) -> ()
   variants: method
 
+# Sets the tensor data held by this `Variable` to be the same as `new_data`.
```

Comment:
I think we should deprecate this in C++ (i.e., only in documentation).

You can say use something like NoGrad and in-place functions instead.

Reply:
Well, in Python I can say `x.data = ...` and to the best of my knowledge we haven't deprecated that yet. Are you suggesting we deprecate this too?

Reply:
YES ! Please !
(note that might require update to some of our optimizers)